pkg/daemon/update.go

```diff
@@ -382,6 +389,49 @@ func verifyUserFields(pwdUser igntypes.PasswdUser) error {
 	return nil
 }
 
+// generateKargsCommand performs a diff between the old/new MC kernelArguments,
+// and generates the command line arguments suitable for `rpm-ostree kargs`.
+// Note what we really should be doing though is also looking at the *current*
+// kernel arguments in case there was drift.  But doing that requires us knowing
+// what the "base" arguments are.  See https://github.com/ostreedev/ostree/issues/479
+func generateKargsCommand(oldConfig, newConfig *mcfgv1.MachineConfig) []string {
+	oldKargs := make(map[string]bool)
+	for _, arg := range oldConfig.Spec.KernelArguments {
+		oldKargs[arg] = true
+	}
+	newKargs := make(map[string]bool)
+	for _, arg := range newConfig.Spec.KernelArguments {
+		newKargs[arg] = true
+	}
+	cmdArgs := []string{}
+	for _, arg := range oldConfig.Spec.KernelArguments {
+		if !newKargs[arg] {
+			cmdArgs = append(cmdArgs, "--delete=" + arg)
+		}
+	}
+	for _, arg := range newConfig.Spec.KernelArguments {
+		if !oldKargs[arg] {
+			cmdArgs = append(cmdArgs, "--append=" + arg)
+		}
+	}
+	return cmdArgs
+}
+
+// updateKernelArguments adjusts the kernel args
+func (dn *Daemon) updateKernelArguments(oldConfig, newConfig *mcfgv1.MachineConfig) error {
+	diff := generateKargsCommand(oldConfig, newConfig)
+	if len(diff) == 0 {
+		return nil
+	}
+	if dn.OperatingSystem != machineConfigDaemonOSRHCOS {
+		return fmt.Errorf("Updating kargs on non-RHCOS nodes is not supported: %v", diff)
```

Comment:
now that I'm looking at this, should we just return nicely as we do for `updateOS` :thinking:

Reply:
https://github.com/openshift/machine-config-operator/pull/824